File: apps/runner-go/internal/permissions/policy.go
package permissions

type Profile string

const (
	Readonly        Profile = "readonly"
	WorkspaceWrite  Profile = "workspace_write"
	TrustedInternal Profile = "trusted_internal"
)

type Decision string

const (
	Allow Decision = "allow"
	Ask   Decision = "ask"
	Deny  Decision = "deny"
)

func ShellDecision(profile Profile) Decision {
	switch profile {
	case TrustedInternal, WorkspaceWrite:
		return Ask
	default:
		return Deny
	}
}

File: packages/core/src/permissions/policy.ts
import type { PermissionProfile } from "@fusion-harness/shared";

export type PermissionDecision = "allow" | "ask" | "deny";

export type PermissionPolicy = {
  profile: PermissionProfile;
  filesystemWrite: PermissionDecision;
  shell: PermissionDecision;
  network: PermissionDecision;
};

export function resolvePermissionPolicy(profile: PermissionProfile): PermissionPolicy {
  switch (profile) {
    case "trusted_internal":
      return { profile, filesystemWrite: "allow", shell: "ask", network: "ask" };
    case "workspace_write":
      return { profile, filesystemWrite: "allow", shell: "ask", network: "ask" };
    case "readonly":
      return { profile, filesystemWrite: "deny", shell: "deny", network: "deny" };
  }
}
